### PR TITLE
Docs modifications to use QuadratureTraining instead of GridTraining

### PR DESCRIPTION
**Pull request by sdesai1287**
solved [#634](url)

### CODE REVIEW

File: docs/src/tutorials/low_level.md [hunk 1]
```diff
@@ -35,12 +35,9 @@ bcs = [u(0, x) ~ -sin(pi * x),
 domains = [t ∈ Interval(0.0, 1.0),
     x ∈ Interval(-1.0, 1.0)]
 
-# Discretization
-dx = 0.05
-
 # Neural network
 chain = Lux.Chain(Dense(2, 16, Lux.σ), Dense(16, 16, Lux.σ), Dense(16, 1))
-strategy = NeuralPDE.GridTraining(dx)
+strategy = NeuralPDE.QuadratureTraining
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/low_level.md:40]:
```suggestion
strategy = NeuralPDE.QuadratureTraining()
```

File: docs/src/tutorials/param_estim.md [hunk 2]
```diff
@@ -113,7 +113,7 @@ Then finally defining and optimizing using the `PhysicsInformedNN` interface.
 
 ```@example param_estim
 discretization = NeuralPDE.PhysicsInformedNN([chain1, chain2, chain3],
-                                             NeuralPDE.GridTraining(dt), param_estim = true,
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/param_estim.md:116]: dt is still defined?

File: docs/src/tutorials/constraints.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ function norm_loss_function(phi, θ, p)
 end
 
 discretization = PhysicsInformedNN(chain,
-                                   GridTraining(dx),
+                                   QuadratureTraining(),
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/constraints.md:68]: dx is still defined.

File: docs/src/tutorials/low_level.md (hunk 1 above)
Comment (sdesai1287) [reply docs/src/tutorials/low_level.md:40]: fixed

File: docs/src/tutorials/param_estim.md (hunk 2 above)
Comment (sdesai1287) [reply docs/src/tutorials/param_estim.md:116]: It seems to be used in other places in this file, thats why I didnt remove it

File: docs/src/tutorials/constraints.md (hunk 3 above)
Comment (sdesai1287) [reply docs/src/tutorials/constraints.md:68]: same as dt from the other file, it seems to be used in other places in this file, thats why I didnt remove it

File: docs/src/tutorials/param_estim.md (hunk 2 above)
Comment (sathvikbhagavan) [reply docs/src/tutorials/param_estim.md:116]: yes, `dt` is used for evenly spaced time grid for the data

File: docs/src/tutorials/constraints.md (hunk 3 above)
Comment (ChrisRackauckas) [reply docs/src/tutorials/constraints.md:68]: But none of this file should be using it anymore?

File: docs/src/tutorials/param_estim.md (hunk 2 above)
Comment (sathvikbhagavan) [reply docs/src/tutorials/param_estim.md:116]: yes, it can be removed

File: docs/src/tutorials/constraints.md (hunk 3 above)
Comment (sdesai1287) [reply docs/src/tutorials/constraints.md:68]: Ok I removed the instances of dx but it seems like the program is still creating a grid of some kind (see line 61 and below), maybe only for evaluation. Do I want to delete all that stuff?

**Comment by sathvikbhagavan** [comment]
Doc build is failing [here](https://github.com/SciML/NeuralPDE.jl/actions/runs/6087807224/job/16517118210?pr=729#step:5:151) and [here](https://github.com/SciML/NeuralPDE.jl/actions/runs/6087807224/job/16517118210?pr=729#step:5:173)

**Comment by sathvikbhagavan** [comment]
Docs is still erroring at https://github.com/SciML/NeuralPDE.jl/actions/runs/6185760782/job/16791965357?pr=729#step:5:86 
The doc build is using ComponentArrays@0.14.1
Any idea why it is failing? Does it need ComponentArrays@0.15?

**Comment by ChrisRackauckas** [comment]
It at least wouldn't need a bump. What was the last version it was using wherei t was passing?

**Comment by sdesai1287** [comment]
This is passing for me locally without bringing up the ComponentArrays issue. Can we rebuild the CI or something? The the changes I made should not affect ComponentArrays

**Comment by ChrisRackauckas** [comment]
I rebased onto master. Let's see how that goes.

**Comment by ChrisRackauckas** [comment]
Just please set the ComponentArrays bound to what it needs to be so we can merge.

**Comment by sathvikbhagavan** [comment]
I am not sure whats happening but I tried building the docs locally and it fails with the same error. But if I tried reducing the maxiters to 1200 from 2000 in https://github.com/SciML/NeuralPDE.jl/blob/master/docs/src/tutorials/constraints.md?plain=1#L91, it surprisingly passes.

**Comment by sathvikbhagavan** [comment]
The docs build is now failing because of  `IntegralsCubature` 0.2.3 [also reproduced it locally] but it is passing with 0.2.2 locally. I haven't looked into the root cause.

The compat for ComponentArrays introduced in [87433bc](https://github.com/SciML/NeuralPDE.jl/pull/729/commits/87433bc079abf4343afd56f7c5b4402d1a972225)  isn't really affecting as it still uses 0.15.3 - https://github.com/SciML/NeuralPDE.jl/actions/runs/6323742800/job/17171915807?pr=729#step:4:96 but the other error we were seeing before is now passing.

**Comment by sathvikbhagavan** [comment]
@ChrisRackauckas any idea why the doc build is erroring with https://github.com/SciML/NeuralPDE.jl/actions/runs/6323742800/job/17171915807?pr=729#step:5:43

I traced it to https://github.com/SciML/Integrals.jl/blob/master/lib/IntegralsCubature/src/IntegralsCubature.jl#L171 locally but I am not what is the exact reason.

**Comment by sathvikbhagavan** [comment]
@sdesai1287 can you rebase your branch on master? Compat bounds were introduced for `IntegralsCubature` in https://github.com/SciML/NeuralPDE.jl/pull/743 where docs passed. It should pass here as well and then this can be merged.

**Comment by ChrisRackauckas** [comment]
The docs successfully built on master, so I put this on top of a successful master.

**Review by sathvikbhagavan — APPROVED** [review]
@ChrisRackauckas docs build is passing. I think we can merge this.

**Comment by ChrisRackauckas** [comment]
🎉

**Pull request merged**